docs/architecture/ADR-004-reserves.md

```diff
@@ -0,0 +1,129 @@
+# ADR 004: Accumulating reserves from borrow interest
+
+## Changelog
+
+- October 14, 2021: Initial Draft (@toteki)
+
+## Status
+
+Proposed
+
+## Context
+
+Borrow positions on Umee accrue interest over time.
+When interest accrues, the sum of all assets owed by all users increases for each borrowed token denomination, and the amount of that increase serves to benefit lenders (by increasing the total amount of assets owed by the Umee system to lenders), and also to increase the amount of base assets the Umee system holds in reserve.
+
+The mechanism by which interest is calculated, and then split between incentivizing lenders as per [ADR-001](./ADR-001-interest-stream.md) and reserves as defined in this ADR, will follow.
+
+## Alternative Approaches
+
+- Setting aside a separate account (or even separate module account, for example by making `x/reserve`) would separate reserve balances from the lending pool.
```

Comment:
nit: I don't think we need to add this section. Alternative approaches are really only needed when there are solid tangible alternative approaches that we considered.

```suggestion
```